src/raftstore/coprocessor/split_observer.rs

```diff
@@ -156,6 +157,28 @@ mod test {
     }
 
     #[test]
+    fn test_forget_encode() {
+        let region_start_key = vec![128, 0, 0, 0, 0, 0, 1, 255, 18, 95, 114, 128, 0, 0, 0, 0, 255,
+                                    0, 16, 154, 0, 0, 0, 0, 0, 250];
+        let key = vec![128, 0, 0, 0, 0, 0, 1, 255, 18, 95, 114, 128, 0, 0, 0, 0, 255, 0, 36, 130,
```

Comment:
No. The origin bug is comparing an encoded start_key with an decoded key.  We should compare them both encoded.